services/frontmatter_generator.py

```
@@ -2,503 +2,255 @@
-                    if answer:
-                        faq_entries.append({
-                            "@type": "Question",
-                            "name": question,
-                            "acceptedAnswer": {
-                                "@type": "Answer",
-                                "text": answer
-                            }
-                        })
+        return s or "post"
+
+    def generate_canonical_url(self, title: str, date_str: str, existing_slugs: List[str] = None) -> str:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)